### PR TITLE
[NP-10138] Cron - add migration support for legacy/deprecated specifies such as 'month'. 

### PR DESCRIPTION
**Pull request by jlhughes**
- add missing legacy javaSetters. Some had been provided but a few were missing. 
- add tests for dayOfWeek and monthOfYear
- Use new Schedule for each test rather than figure out what needs to be reset.

### CODE REVIEW

File: src/foam/nanos/cron/CronSchedule.js [hunk 1]
```diff
@@ -93,6 +93,18 @@ foam.CLASS({
...
       documentation: 'comma seperated hours',
     },
+    {
+      documentation: 'deprecated, replaced by monthsOfYear',
+      class: 'Int',
+      name: 'month',
+      transient: true,
+      hidden: true,
+      javaSetter: `
+      if ( val > 0 && ! monthsOfYearIsSet_ ) {
```
Comment (jinnnn9) [inline src/foam/nanos/cron/CronSchedule.js]: Do we want to append the value to the array if it is already defined and possibly has other values in it? Same question for all the java setters supporting deprecated properties.

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: No.  If there is already a value then the new property is being used.

Comment (jinnnn9) [reply src/foam/nanos/cron/CronSchedule.js]: if month is -1, shouldn't we add every month to monthsOfYear?

**Comment by jinnnn9** [comment]
Getting class cast exception when trying to update a cron that has dayOfMonth. Investigating.

Update: fixed

File: src/foam/nanos/cron/CronSchedule.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: Yes, since -1 was wildcard. Need to update all.

File: src/foam/nanos/cron/CronSchedule.js [hunk 2]
```diff
@@ -52,12 +56,13 @@ foam.CLASS({
     {
       class: 'Int',
       name: 'second',
-      value: 0,
+      value: -1,
```
Comment (jinnnn9) [inline src/foam/nanos/cron/CronSchedule.js]: does this contradict with min, max?

File: src/foam/nanos/cron/CronSchedule.js [hunk 3]
```diff
@@ -143,17 +180,29 @@ foam.CLASS({
...
       }
     },
+    {
+      documentation: 'deprecated, replaced by daysOfMonth',
+      class: 'Int',
+      name: 'dayOfMonth',
+      transient: true,
+      hidden: true,
+      javaSetter: `
+      if ( val > 0 &&
```
Comment (jinnnn9) [inline src/foam/nanos/cron/CronSchedule.js:201]: dayOfMonth can be -1

File: src/foam/nanos/cron/CronSchedule.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js:59]: please elaborate.

File: src/foam/nanos/cron/CronSchedule.js (hunk 3 above)
Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js:201]: No cronjobs use it, and the default is to run everyday, so no need to support legacy jobs. 
dayOfWeek/daysOfWeek dayOfMonth/daysOfMonth are mutually exclusive. 
So previously either being -1 would be everyday.

File: src/foam/nanos/cron/CronSchedule.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: managed to revert default seconds to 0.

Comment (jinnnn9) [reply src/foam/nanos/cron/CronSchedule.js]: We still have some cron jobs with second set to -1. Did this mean that the cron should run every second? In any case, don't we need something like javaPreset to adjust the second when it was set to -1 already?

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: we don't want anything running every second and I'm sure those same jobs have some other specifier set. So we are not supporting every second.

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: Additionally, for schedules that run every 5 minutes, for example, then the IntervalScheduler should be used.  The CronScheduler is intended for jobs that run once or twice a day with additional week day, month, ... requirements.

Comment (jinnnn9) [reply src/foam/nanos/cron/CronSchedule.js]: Yes those crons have other time specifiers set. As long as you are okay with them having second set to -1 (though it doesn't mean anything and contradict with min max on second property), the pr looks good to me

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js]: Thanks. Yes, -1 is ignored in getNextScheduledTime.

**Pull request merged**